torch/utils/tensorboard/pytorch_graph.py

```diff
@@ -0,0 +1,274 @@
+        return '\n'.join(repr) + '\n\n'
+
+
+class Node_py(Node_base):
+    def __init__(self, Node_cpp, valid_mothods):
+        super(Node_py, self).__init__(Node_cpp)
+        self.valid_mothods = valid_mothods[:]
+        self.inputs = []
+
+        for m in self.valid_mothods:
+            if m == 'inputs' or m == 'outputs':
+                list_of_node = list(getattr(Node_cpp, m)())
+                io_uniqueName_list = []
+                io_tensorSize_list = []
+                for n in list_of_node:
+                    io_uniqueName_list.append(n.uniqueName())
+                    if n.type().kind() == 'CompleteTensorType':
+                        io_tensorSize_list.append(n.type().sizes())
+                    else:
+                        io_tensorSize_list.append(None)
+
+                setattr(self, m, io_uniqueName_list)
+                setattr(self, m + 'TensorSize', io_tensorSize_list)
+
+            else:
+                setattr(self, m, getattr(Node_cpp, m)())
+
+
+class Node_py_IO(Node_py):
+    def __init__(self, Node_cpp, input_or_output=None):
+        super(Node_py_IO, self).__init__(Node_cpp, methods_IO)
+        try:
+            tensorsize = Node_cpp.type().sizes()
+        except RuntimeError:
+            tensorsize = [1, ]  # fail when constant model is used.
+        self.tensorSize = tensorsize
+        self.kind = 'Parameter'
+        if input_or_output:
+            self.input_or_output = input_or_output
+            self.kind = 'IO Node'
+
+
+class Node_py_OP(Node_py):
+    def __init__(self, Node_cpp):
+        super(Node_py_OP, self).__init__(Node_cpp, methods_OP)
+        self.attributes = str({k: Node_cpp[k] for k in Node_cpp.attributeNames()}).replace("'", ' ')
+        self.kind = Node_cpp.kind()
+
+
+class Graph_py(object):
+    def __init__(self):
+        self.nodes_OP = []
+        self.nodes_IO = OrderedDict()
+        self.uniqueNameToScopedName = {}
+        self.shallowestScopeName = 'default'
+        self.scope_name_appeared = []
+
+    def append(self, x):
+        if type(x) == Node_py_IO:
+            self.nodes_IO[x.uniqueName] = x
+        if type(x) == Node_py_OP:
+            self.nodes_OP.append(x)
+            for node_output, outputSize in zip(x.outputs, x.outputsTensorSize):
+                self.scope_name_appeared.append(x.scopeName)
+                self.nodes_IO[node_output] = Node_base(node_output,
+                                                       x.inputs,
+                                                       x.scopeName,
+                                                       outputSize,
+                                                       op_type=x.kind,
+                                                       attributes=x.attributes)
+
+    def printall(self):
+        print('all nodes')
+        for node in self.nodes_OP:
+            print(node)
+        for key in self.nodes_IO:
+            print(self.nodes_IO[key])
+
+    def findCommonRoot(self):
```

Comment:
This method name is inconsistent with PEP naming convention, and also the rest of this file too.

Reply:
Good point - corrected naming here.